### ISSUE TITLE
[Bug]: Open favorite folder from sidebar shows root folder content (again)

### ISSUE THREAD

**Issue by steffen-kdab** [nextcloud/server#41007]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

A bug with the exact same symptoms as https://github.com/nextcloud/server/issues/12493
is back in Nextcloud 26.0.6+, and reproducible in 27.1.2 : clicking on a favorite entry, a recent 
entry, a search result for a Label etc. ends up showing the listing of the root folder instead. 

Tested in Firefox 118 Windows, Firefox 118 Linux, and Chromium 117.0.5938.132 
Linux, Chrome 118.0.5993.88 (Official Build)  Linux

### Steps to reproduce

1. have folder `abc`
2. mark it as favorite `abc`
3. Click Favorites in the left sidebar to open Favorites
4. click on the item `abc` in the main view/file list
5. bug: we see the content of the home/root folder in the file list

### Expected behavior

We should see content of `abc` in the file list

### Installation method

Community Docker image

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "27.1.2.1",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_sendmailmode": "smtp",
        "mail_smtpport": "25",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "app_install_overwrite": [
            "ldapcontacts",
            "groupfolders"
        ],
        "overwritehost": "[REDACTED]",
        "overwriteprotocol": "https",
        "mail_smtpsecure": "tls",
        "loglevel": 0,
        "debug": false,
        "theme": "",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "default_phone_region": "SE"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.19.0
  - admin_audit: 1.17.0
  - analytics: 4.11.0
  - calendar: 4.5.2
  - camerarawpreviews: 0.8.2
  - circles: 27.0.1
  - cloud_federation_api: 1.10.0
  - comments: 1.17.0
  - contacts: 5.4.2
  - contactsinteraction: 1.8.0
  - dashboard: 7.7.0
  - dav: 1.27.0
  - federatedfilesharing: 1.17.0
  - federation: 1.17.0
  - files: 1.22.0
  - files_accesscontrol: 1.17.1
  - files_external: 1.19.0
  - files_pdfviewer: 2.8.0
  - files_reminders: 1.0.0
  - files_rightclick: 1.6.0
  - files_sharing: 1.19.0
  - files_trashbin: 1.17.0
  - files_versions: 1.20.0
  - firstrunwizard: 2.16.0
  - groupfolders: 15.3.1
  - impersonate: 1.14.0
  - logreader: 2.12.0
  - lookup_server_connector: 1.15.0
  - mail: 3.4.2
  - maps: 1.1.1
  - nextcloud_announcements: 1.16.0
  - notifications: 2.15.0
  - oauth2: 1.15.1
  - password_policy: 1.17.0
  - photos: 2.3.0
  - polls: 5.3.2
  - previewgenerator: 5.3.0
  - privacy: 1.11.0
  - provisioning_api: 1.17.0
  - recommendations: 1.6.0
  - related_resources: 1.2.0
  - serverinfo: 1.17.0
  - settings: 1.9.0
  - sharebymail: 1.17.0
  - support: 1.10.0
  - survey_client: 1.15.0
  - systemtags: 1.17.0
  - tasks: 0.15.0
  - text: 3.8.0
  - theming: 2.2.0
  - twofactor_backupcodes: 1.16.0
  - updatenotification: 1.17.0
  - user_ldap: 1.17.0
  - user_status: 1.7.0
  - viewer: 2.1.0
  - weather_status: 1.7.0
  - workflowengine: 2.9.0
Disabled:
  - bruteforcesettings: 2.7.0
  - encryption: 2.15.0
  - files_markdown: 2.4.1 (installed 2.4.1)
  - ldap_contacts_backend: 1.7.0 (installed 1.7.0)
  - recognize: 5.0.3 (installed 5.0.3)
  - secrets: 1.5.2 (installed 1.5.2)
  - sharerenamer: 3.2.0 (installed 3.2.0)
  - spreed: 17.1.1 (installed 17.1.1)
  - suspicious_login: 5.0.0
  - twofactor_totp: 9.0.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
Info	text	Keep previous document of 291106		2023-10-20T00:26:49+0200
Info	text	Keep previous document of 291106		2023-10-20T00:26:00+0200
Warning	OC\Log\Rotate	Log file "/var/www/html/data/nextcloud.log" was over 104857600 bytes, moved to "/var/www/html/data/nextcloud.log.1"		2023-10-20T00:25:01+0200
```

### Additional info

We have an nginx reverse proxy in front of the docker container to provide HTTPS etc:

```
server {
    listen 80;
    listen [::]:80;

    # listen 443 ssl http2;
    # listen [::]:443 ssl http2;
    listen 443 ssl;
    listen [::]:443 ssl;

    ssl_certificate /etc/letsencrypt/live/[REDACTED]/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/[REDACTED]/privkey.pem; # managed by Certbot

    server_name ***REMOVED SENSITIVE VALUE***;

    access_log /var/log/nginx/[REDACTED]-access.log;
    error_log /var/log/nginx/[REDACTED]-error.log ;

    if ( $https != on ) { return 301 https://$host$request_uri; }

    # reverse proxy settings common to all locations
    proxy_set_header Upgrade           $http_upgrade;
    proxy_set_header Connection        "upgrade";
    proxy_set_header Host $host;
    proxy_set_header X-Forwarded-For   $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-Host  $host;
    proxy_set_header X-Forwarded-Port  $server_port;
    proxy_set_header X-Real-IP         $remote_addr;
    proxy_buffering  off;
    add_header Strict-Transport-Security "max-age=15552000; includeSubDomains" always;

    # Support large uploads that take time...
    proxy_connect_timeout 60m;
    proxy_send_timeout 60m;
    proxy_read_timeout 600m;
    proxy_request_buffering off;
    client_max_body_size    0;
    client_body_buffer_size 400M;
    proxy_http_version 1.1;
    proxy_buffers 64 128k;
    proxy_max_temp_file_size 0;
    large_client_header_buffers 16 32k;

    location / {
        proxy_pass       http://192.168.122.135:8080;
    }

    location /.well-known/carddav {
        return 301 $scheme://$host/remote.php/dav;
    }

    location /.well-known/caldav {
        return 301 $scheme://$host/remote.php/dav;
    }

#   location /404.html { internal; }

}
```

**Comment by szaimen** [comment]
cc @skjnldsv

**Comment by ShGKme** [comment]
I cannot reproduce it in 27.1.3.

Though it looks similar to https://github.com/nextcloud/server/issues/39565, I don't the same problem with race condition in HTTP requests.

If anyone can reproduce it, could you check:
1. it is also reproducible on the latest `stable27`
2. what network requests there are on the click, there are too many different requests. If there are, what initiator it has in the devtools.

**Comment by ShGKme** [comment]
Reproduces with the help of @mahibi 

According to the network, there is the same race condition that was in #39565
![image](https://github.com/nextcloud/server/assets/25978914/79951bce-05ec-4700-996e-82151582e51d)

It is fixed and is to be released in 27.1.4
